convert2rhel/main.py

```diff
@@ -63,15 +64,34 @@ def initialize_logger(log_name, log_dir):
 
 
 def main():
-    """Perform all steps for the entire conversion process."""
+    """Wrapper around the main function; performs everything we have
+    to set up before starting the actual conversion process itself,
+    then calls main_locked()--protected by the application lock--to do
+    the conversion process.
+    """
 
     process_phase = ConversionPhase.INIT
 
+    # handle command line arguments
+    toolopts.CLI()
+
     # initialize logging
     initialize_logger("convert2rhel.log", logger_module.LOG_DIR)
 
-    # handle command line arguments
-    toolopts.CLI()
```

Comment:
Joe and I talked on slack.  The traceback is happening when `initialize_logger()` is called because it needs root permissions on the log directory to rotate the log files.  

I think the solution is to move `require_root()` above `initialize_logger()` (previous to the applock PR, it was the first thing we did once we entered `main()`) .  `toolopts.CLI()` should be moved back to after `initialize_logger()` for this PR.

Reply:
Done.